addons/mass_mailing/static/src/interactions/subscribe.js

```diff
@@ -0,0 +1,346 @@
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+import { rpc } from "@web/core/network/rpc";
+import { setElementContent } from "@web/core/utils/html";
+import { renderToMarkup, renderToFragment } from "@web/core/utils/render";
+import { _t } from "@web/core/l10n/translation";
+
+export class Subscribe extends Interaction {
+    static selector = "#o_mailing_portal_subscription";
+    dynamicContent = {
+        "#button_blocklist_add": {
+            "t-on-click.prevent": this.onBlocklistAddClick,
+            "t-att-class": () => ({
+                "d-none":
+                    !this.customerData.blocklistEnabled ||
+                    !this.customerData.blocklistPossible ||
+                    this.customerData.isBlocklisted,
+            }),
+        },
+        "#button_blocklist_remove": {
+            "t-on-click.prevent": this.onBlocklistRemoveClick,
+            "t-att-class": () => ({
+                "d-none": !this.customerData.isBlocklisted,
+            }),
+        },
+        "#button_form_send": {
+            "t-on-click.prevent": this.onFormSend,
+            "t-att-class": () => ({
+                "d-none": this.isFormReadonly,
+            }),
+            "t-att-disabled": () => this.isFormReadonly,
+        },
+        "#button_feedback": {
+            "t-on-click.prevent": this.onFeedbackClick,
+            "t-att-disabled": () => this.isFeedbackReadOnly,
+        },
+        ".o_mailing_subscription_opt_out_reason": {
+            "t-on-click": this.onOptOutReasonClick,
+        },
+        "div#o_mailing_subscription_feedback textarea": {
+            "t-att-class": () => ({
+                "d-none": !this.showFeedbackTextbox,
+            }),
+            "t-att-disabled": () => this.isFeedbackReadOnly,
+        },
+        "div#o_mailing_subscription_feedback input": {
+            "t-att-disabled": () => this.isFeedbackReadOnly,
+        },
```

Comment:
I guess it's ordered alphabetically but it would be nice to keep related declarations together so that we can look at them as blocks in their hierarchical order. Here for example we have `#o_mailing_subscription_feedback` a bit below, would use the same selector here and there so that we can put first the parent, then the children.

Arguably at the risk of making selectors longer to clarify their relations.
Alternatively could separate into sections with comments to more or less match the original separation in 4 files.